server/src/main/java/org/elasticsearch/index/engine/CombinedDeletionPolicy.java

```diff
@@ -242,6 +257,9 @@ synchronized boolean releaseCommit(final IndexCommit acquiredCommit) {
             }
             return count - 1;
         });
+        if (snapshotIndexCommit.acquiredByCommitsListener) {
+            indexCommitsAcquiredByCommitsListener.remove(releasingCommit);
```

Comment:
Can we assert that it got removed?

Reply:
I've changed `indexCommitsAcquiredByCommitsListener` to be `Map<IndexCommit, Integer>` as `externallyAcquiredIndexCommits`, so we piggy back on the exisiting checks that the released commit.